packages/grid/_modules_/grid/hooks/features/filter/useGridFilter.ts

```diff
@@ -441,4 +408,22 @@ export const useGridFilter = (
     apiRef.current.unstable_applyFilters,
   );
   useGridApiEventHandler(apiRef, GridEvents.columnsChange, handleColumnsChange);
+  useGridApiEventHandler(apiRef, GridEvents.preProcessorRegister, handlePreProcessorRegister);
+
+  /**
+   * EFFECTS
+   */
+  useFirstRender(() => {
+    filteringMethodCollectionRef.current = apiRef.current.unstable_applyPreProcessors(
```

Comment:
This code is after the `useGridRegisterFilteringMethod(apiRef, 'none', flatFilteringMethod);` so in flat mode there is always a filtering algorithm defined when applying the filters for the 1st time.

Reply:
This should be a comment in the code so we know to never move it out of this position.